### ISSUE TITLE
Go to Definition nor hover working

### ISSUE THREAD

**Issue by contentfree** [castwide/vscode-solargraph#32]
Using VS Code 1.21.0 on macOS, Ruby extension 0.17.0 and Ruby Solargraph extension 0.17.3, nothing is working anymore.

1. Hovering over things produces no visual change besides flashing "Loading" very quickly. The console states:
```
ERR An unknown error occurred. Please consult the log for more details.  log.ts:171
[Extension Host] 127.0.0.1 - - [12/Mar/2018:14:02:55 PDT] "POST /hover HTTP/1.1" 200 32  console.ts:136 
- -> /hover
```

2. Right-clicking a class (of mine) and choosing `Go to Definition` animates a narrow blue line beneath the tabs for a second or so then shows a `No definition found for 'MyClass'` tooltip. The console states:
```
ERR An unknown error occurred. Please consult the log for more details.  log.ts:171
[Extension Host] 127.0.0.1 - - [12/Mar/2018:14:06:07 PDT] "POST /hover HTTP/1.1" 200 32  console.ts:136 
- -> /hover
```

What log am I supposed to consult for more details (and so I can provide them here)?

**Comment by castwide** [comment]
I just tested a small project on a MacBook by opening a folder with two Ruby files, one containing a class definition and the other using an instance of the class. Hover and go to definition both worked.

I'm not sure what to make of the `unknown error` message. I can't even confirm whether the error came from this extension or not. I've seen it happen for a few different reasons, and finding the root cause has always been a crapshoot. As far as I can tell, the suggestion to `consult the log` is misleading. I don't think VS Code logs anything outside of the console. (If anyone knows of a separate log file or something, please let me know.)

A few questions that might help with troubleshooting:

* What version of the solargraph gem are you running? You can check with `solargraph -v`. Latest is 0.17.3.
* Are there any other errors in the console? The server usually provides a stack trace if it throws an exception.
* Does it work if you disable the Ruby extension?
* Are you using a ruby version manager like rvm or rbenv?
* Do you have a folder open or are you opening individual files?
* If you have a folder open, how large is the project?
* Are you using a .solargraph.yml file?
* Have you changed any configuration settings, such as `solargraph.commandPath` or `solargraph.useBundler`?

**Comment by contentfree** [comment]
Answers for you:

1. Solargraph version: 0.17.3 (yes, extension version is 0.13.0)
2. No stack trace in the console. (However, upon reloading the window I've seen logs at the end of this update)
3. It still fails to work with the Ruby extension disabled
4. I'm using RVM with Ruby 2.3.4
5. I have a folder open
6. It's not a large project (78MB ~100k lines all-in including whitespace and non-code files … basically the result of `git ls-files | xargs wc -l`)
7. I'm not (knowingly) using a .solargraph.yml file
8. I've set `solargraph.commandPath` to the path to the solargraph binary in a specific ruby version in rvm: `/Users/dave/.rvm/gems/ruby-2.3.4/bin/solargraph`

Hopefully something there and in the logs below will point us in a direction.

# 
```
console.ts:136 [Extension Host] Starting the server
console.ts:136 [Extension Host] [2018-03-13 16:42:28] INFO  WEBrick 1.3.1
[2018-03-13 16:42:28] INFO  ruby 2.3.4 (2017-03-30) [x86_64-darwin16]

console.ts:136 [Extension Host] == Sinatra (v2.0.0) has taken the stage on 49418 for development with backup from WEBrick

console.ts:136 [Extension Host] [2018-03-13 16:42:28] INFO  WEBrick::HTTPServer#start: pid=36610 port=49418

console.ts:136 [Extension Host] Preparing /Users/dave/Sites/my-project

console.ts:136 [Extension Host] warning: parser/current is loading parser/ruby23, which recognizes
warning: 2.3.5-compliant syntax, but you are running 2.3.4.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.

console.ts:136 [Extension Host] invalid byte sequence in US-ASCII

console.ts:136 [Extension Host] /Users/dave/.rvm/gems/ruby-2.3.4/gems/solargraph-0.17.3/lib/solargraph/api_map/source.rb:384:in `gsub'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/solargraph-0.17.3/lib/solargraph/api_map/source.rb:384:in `load'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/solargraph-0.17.3/lib/solargraph/api_map.rb:34:in `block in initialize'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/solargraph-0.17.3/lib/solargraph/api_map.rb:32:in `each'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/solargraph-0.17.3/lib/solargraph/api_map.rb:32:in `initialize'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/solargraph-0.17.3/lib/solargraph/server.rb:209:in `new'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/solargraph-0.17.3/lib/solargraph/server.rb:209:in `generate_api_map'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/solargraph-0.17.3/lib/solargraph/server.rb:204:in `prepare_workspace'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/solargraph-0.17.3/lib/solargraph/server.rb:26:in `block in <class:Server>'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1632:in `call'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1632:in `block in compile!'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:991:in `block (3 levels) in route!'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1010:in `route_eval'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:991:in `block (2 levels) in route!'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1037:in `block in process_route'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1035:in `catch'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1035:in `process_route'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:989:in `block in route!'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:988:in `each'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:988:in `route!'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1094:in `block in dispatch!'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1073:in `block in invoke'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1073:in `catch'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1073:in `invoke'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1091:in `dispatch!'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:923:in `block in call!'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1073:in `block in invoke'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1073:in `catch'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1073:in `invoke'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:923:in `call!'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:913:in `call'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/rack-protection-2.0.0/lib/rack/protection/xss_header.rb:18:in `call'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/rack-protection-2.0.0/lib/rack/protection/path_traversal.rb:16:in `call'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/rack-protection-2.0.0/lib/rack/protection/json_csrf.rb:26:in `call'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/rack-protection-2.0.0/lib/rack/protection/base.rb:50:in `call'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/rack-protection-2.0.0/lib/rack/protection/base.rb:50:in `call'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/rack-protection-2.0.0/lib/rack/protection/frame_options.rb:31:in `call'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/rack-2.0.4/lib/rack/null_logger.rb:9:in `call'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/rack-2.0.4/lib/rack/head.rb:12:in `call'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/show_exceptions.rb:22:in `call'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:194:in `call'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1955:in `call'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1499:in `block in call'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1726:in `synchronize'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/sinatra-2.0.0/lib/sinatra/base.rb:1499:in `call'
/Users/dave/.rvm/gems/ruby-2.3.4/gems/rack-2.0.4/lib/rack/handler/webrick.rb:86:in `service'
/Users/dave/.rvm/rubies/ruby-2.3.4/lib/ruby/2.3.0/webrick/httpserver.rb:140:in `service'
/Users/dave/.rvm/rubies/ruby-2.3.4/lib/ruby/2.3.0/webrick/httpserver.rb:96:in `run'
/Users/dave/.rvm/rubies/ruby-2.3.4/lib/ruby/2.3.0/webrick/server.rb:296:in `block in start_thread'

console.ts:136 [Extension Host] 127.0.0.1 - - [13/Mar/2018:16:42:28 PDT] "POST /prepare HTTP/1.1" 200 5698
- -> /prepare
```

**Comment by castwide** [comment]
Thanks for the info. The log is especially helpful.

The procedure that creates the initial workspace map is failing to parse a file due to a character encoding issue. I've seen similar problems with files that contain Unicode sequences. Unfortunately, the error unfortunately doesn't tell us which file failed.

I'll work on a fix for the next release of the gem. At the very least, it should log an error with the filename that failed to parse and continue mapping the rest of the workspace. It already handles a lot of parsing errors that way, but this one must have gotten overlooked.

**Comment by contentfree** [comment]
Cool. Please let me know if I can help

**Comment by JeffLuckett** [comment]
Just chiming in that I've got the same issue.

OSX: 10.13.3
VSCode: 1.21.1
Solargraph: 0.17.3
Solargraph Plugin: 0.13.0
rebornix.ruby: 0.17.0
Ruby: 2.4.1

Other than the fully qualified paths, the stack trace is pretty much identical to the one @contentfree posted [above](https://github.com/castwide/vscode-solargraph/issues/32#issuecomment-372858258).

I noticed it had stopped working a little while back, but was too busy to track it down. It bugged me enough today 😄 .
